### ISSUE TITLE
[BUG] ControlledQubitUnitary decomposes to itself

### ISSUE THREAD

**Issue by komiamiko** [PennyLaneAI/pennylane#3878]
### Expected behavior

`ControlledQubitUnitary` should decompose to something else ex. ZYZ decomposition

### Actual behavior

`ControlledQubitUnitary` decomposes to itself

### Additional information

Already fixed in https://github.com/PennyLaneAI/pennylane/pull/3851

### Source code

```shell
>>> import pennylane as qml
>>> x = qml.ControlledQubitUnitary(qml.Hadamard.compute_matrix(), [0], [1])
>>> y = x.decomposition()
>>> x
Controlled(QubitUnitary(array([[ 0.70710678,  0.70710678],
       [ 0.70710678, -0.70710678]]), wires=[1]), control_wires=[0])
>>> y
[Controlled(QubitUnitary(array([[ 0.70710678,  0.70710678],
       [ 0.70710678, -0.70710678]]), wires=[1]), control_wires=[0])]
>>> qml.equal(x, y[0])
True
>>> exit()
```

### System information

```shell
ame: PennyLane
Version: 0.30.0.dev0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /home/ubuntu/anaconda3/envs/plwork/lib/python3.10/site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, retworkx, scipy, semantic-version, toml
Required-by: PennyLane-Lightning

Platform info:           Linux-5.15.0-1030-aws-x86_64-with-glibc2.31
Python version:          3.10.9
Numpy version:           1.23.5
Scipy version:           1.10.1
Installed devices:
- default.gaussian (PennyLane-0.30.0.dev0)
- default.mixed (PennyLane-0.30.0.dev0)
- default.qubit (PennyLane-0.30.0.dev0)
- default.qubit.autograd (PennyLane-0.30.0.dev0)
- default.qubit.jax (PennyLane-0.30.0.dev0)
- default.qubit.tf (PennyLane-0.30.0.dev0)
- default.qubit.torch (PennyLane-0.30.0.dev0)
- default.qutrit (PennyLane-0.30.0.dev0)
- null.qubit (PennyLane-0.30.0.dev0)
- lightning.qubit (PennyLane-Lightning-0.29.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by komiamiko** [comment]
This was discovered in the process of making https://github.com/PennyLaneAI/pennylane/pull/3851 when the test got stuck in an infinite loop. https://github.com/PennyLaneAI/pennylane/pull/3851 also fixes it.

**Comment by albi3ro** [comment]
Another option is simply getting rid of `QubitUnitary._controlled`.  Does `ControlledQubitUnitary` give us anything `Controlled(QubitUnitary)` doesn't? Not really.

**Comment by albi3ro** [comment]
This bug has been closed in #3851
